libxrdp/xrdp_sec.c

```diff
@@ -782,20 +782,20 @@ xrdp_sec_process_logon_info(struct xrdp_sec *self, struct stream *s)
         return 1;
     }
 
-    if (unicode_in(s, len_domain, self->rdp_layer->client_info.domain, 255) != 0)
+    if (unicode_in(s, len_domain, INFO_TEXT_MAX_UNICODE_BYTES, self->rdp_layer->client_info.domain, INFO_TEXT_MAX_BYTES) != 0)
```

Comment:
The limit on the output comes from the output buffer size. I would rather use `sizeof(self->rdp_layer->client_info.domain) - 1`, which would show where the limit comes from. INFO_TEXT_MAX_BYTES (if needed at all) should be where `domain` is defined, i.e. xrdp_client_info.h